app/demo-geant-integration/PrimaryGeneratorAction.cc

```diff
@@ -0,0 +1,34 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file demo-geant-integration/PrimaryGeneratorAction.cc
+//---------------------------------------------------------------------------//
+#include "PrimaryGeneratorAction.hh"
+
+#include <G4ParticleGun.hh>
+#include <G4ParticleTable.hh>
+#include <G4SystemOfUnits.hh>
+#include <G4ThreeVector.hh>
+
+namespace demo_geant
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Generate primaries based on the particle gun data.
+ */
+void PrimaryGeneratorAction::GeneratePrimaries(G4Event* event)
+{
+    auto g4particle_def = G4ParticleTable::GetParticleTable()->FindParticle(11);
```

Comment:
use hadrons (pions or protons) and add the number of primary particles to generate if we do not use a realistic event.

Reply:
Yeah, we will be deleting this shortly in favor of a HepMC3 input.